tests/unit/python/fledge/services/core/api/plugins/test_update.py

```diff
@@ -0,0 +1,381 @@
+# -*- coding: utf-8 -*-
+
+# FLEDGE_BEGIN
+# See: http://fledge.readthedocs.io/
+# FLEDGE_END
+
+import json
+import uuid
+from unittest.mock import patch, MagicMock
+import pytest
+
+from aiohttp import web
+
+from fledge.services.core import routes
+from fledge.services.core import server
+from fledge.services.core import connect
+from fledge.services.core.api.plugins import update as plugins_update
+from fledge.services.core.api.plugins.exceptions import *
+from fledge.services.core.scheduler.scheduler import Scheduler
+from fledge.common.storage_client.storage_client import StorageClientAsync
+from fledge.common.plugin_discovery import PluginDiscovery
+from fledge.common.configuration_manager import ConfigurationManager
+
+
+__author__ = "Ashish Jabble"
+__copyright__ = "Copyright (c) 2020 Dianomic Systems Inc."
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+@pytest.allure.feature("unit")
+@pytest.allure.story("api", "plugins", "update")
+class TestPluginUpdate:
+    @pytest.fixture
+    def client(self, loop, test_client):
+        app = web.Application(loop=loop)
+        # fill the routes table
+        routes.setup(app)
+        return loop.run_until_complete(test_client(app))
+
+    @pytest.mark.parametrize("param", [
+        "blah",
+        1,
+        "notificationDelivery"
+        "notificationRule"
+    ])
+    async def test_bad_type_plugin(self, client, param):
+        resp = await client.put('/fledge/plugins/{}/name/update'.format(param), data=None)
+        assert 400 == resp.status
+        assert "Invalid plugin type. Must be 'north' or 'south' or 'filter' or 'notify' or 'rule'" == resp.reason
```

Comment:
May be better message is `Must be one of 'south' , north', 'filter', 'notify' or 'rule'" `